packages/components/plh/parent-group/plh-parent-group.service.ts

```diff
@@ -461,6 +444,40 @@ export class PlhParentGroupService extends SyncServiceBase {
     } as IParentGroup;
   }
 
+  /**
+   * Format parent group data for push to shared data by removing protected fields and RapidPro fields
+   */
+  private formatParentGroupDataForPush(parentGroup: IParentGroup): IParentGroup {
+    // Remove any field whose key starts with "rp_" (e.g. "rp_access_code") or "rapidpro_" (e.g. "rapidpro_uuid")
+    parentGroup = Object.fromEntries(
+      Object.entries(parentGroup).filter(
+        ([key]) => !key.startsWith("rp_") && !key.startsWith("rapidpro_")
+      )
+    ) as IParentGroup;
+
+    parentGroup.parents = parentGroup.parents.map((parent) =>
+      this.hackRemoveRapidProFieldsFromParentData(parent)
+    );
+
+    return parentGroup;
+  }
+
+  /**
+   * Format parent data for upload to shared data by removing RapidPro fields,
+   * since these are managed by RapidPro in shared data
+   */
+  private hackRemoveRapidProFieldsFromParentData(parent: IParent | IParentFromRapidPro): IParent {
+    if (this.isParentFromRapidPro(parent)) {
```

Comment:
nit(non-blocking): Just to check, is the long-term plan to have some parents registered through rapidpro and others from alternate sources? Or is the discrepancy just due to data that's manually been put in the db for testing? Working with one type of data would definitely be easy than multiple, although in this case I think likely the same method can still be ran against any parent data even if not from rapidpro (given you're just filtering out object keys by prefix)

Assuming the service is designed to be closely coupled with rapidpro, I think you can probably remove `hack` from these methods as they are somewhat integral to the service and unlikely to be removed in the future.